src/kafka-util/src/bin/kgen.rs

```diff
@@ -584,16 +615,22 @@ async fn main() -> anyhow::Result<()> {
             eprintln!("Generating message {}", i);
         }
         value_gen.next_value(&mut buf);
-        let key_i = if let Some(key_dist) = key_dist.as_ref() {
-            key_dist.sample(&mut rng)
+        let key = if let Some(key_gen) = key_gen.as_mut() {
+            // TODO: shitty
```

Comment:
I think this `TODO` could be improved with a reason for why this is... in need of improvement.